### PR TITLE
Remove Caching.Redis package

### LINKED ISSUE #423: Remove the old Redis dependency
We're adding a new dependency in #410 for 2.2.

In 3.0, let's remove Microsoft.Extensions.Caching.Redis package.

cc @anurse @Tratcher @DamianEdwards @Eilon

### PR DESCRIPTION
**Pull request by BrennanConroy**
Fixes #423

### CODE REVIEW

File: Caching.sln [hunk 1]
```diff
@@ -1,6 +1,6 @@
 Microsoft Visual Studio Solution File, Format Version 12.00
-# Visual Studio 15
-VisualStudioVersion = 15.0.26730.10
+# Visual Studio Version 16
```
Comment (natemcmaster) [inline Caching.sln]: Let's keep this file compatible with VS 2017 until VS 2019 has a public preview.

**Comment by Eilon** [comment]
DOes anything also need to be removed/updated in some Universe-like place? artifacts.props or something?

**Comment by BrennanConroy** [comment]
Checkout https://github.com/aspnet/AspNetCore/pull/3779

**Comment by Eilon** [comment]
Just saw it, thanx 😄

**Pull request merged**